pkgs/servers/peertube/default.nix

```diff
@@ -0,0 +1,132 @@
+{ light ? null, pkgs, stdenv, lib, fetchurl, fetchFromGitHub, callPackage
+, nodejs ? pkgs.nodejs-14_x
+, jq, youtube-dl, nodePackages, yarn2nix-moretea }:
+
+/*
+updating:
+cd pkgs/servers/peertube
+git clone --depth 1 --branch v3.4.0 https://github.com/Chocobozzz/PeerTube.git /tmp/PeerTube
+(cd /tmp/PeerTube/ && yarn2nix > yarn.nix)
+(cd /tmp/PeerTube/client && yarn2nix > yarn.nix)
+cp /tmp/PeerTube/package.json .
+cp /tmp/PeerTube/yarn.lock .
+cp /tmp/PeerTube/yarn.nix .
+mkdir -p client
+cp /tmp/PeerTube/client/package.json client/
+cp /tmp/PeerTube/client/yarn.lock client/
+cp /tmp/PeerTube/client/yarn.nix client/
+*/
+let
+  version = "3.4.0";
+
+  source = fetchFromGitHub {
+    owner = "Chocobozzz";
+    repo = "PeerTube";
+    rev = "v${version}";
+    sha256 = "sha256-cYmdVBx3XBMAUWMJ7DqFtlSwLe5g6mPFPUN8GVGuHIc=";
+  };
+
+  patchedSource = stdenv.mkDerivation {
+    pname = "peertube";
+    inherit version;
+    src = source;
+    phases = [ "unpackPhase" "patchPhase" "installPhase" ];
+    patches = [ ./fix_yarn_lock.patch ];
+    installPhase = ''
+      mkdir $out
+      cp -a . $out/
+    '';
+  };
```

Comment:
```suggestion
```

I've just confirmed, if you use a yarn.nix file, the patching of the yarn.lock is not necessary.
tested on darwin.